zilliqa/src/api/zil.rs

```diff
@@ -341,8 +342,13 @@ fn get_smart_contract_code(params: Params, node: &Arc<Mutex<Node>>) -> Result<Va
         .get_state(&block)?
         .get_account(smart_contract_address)?;
 
-    let Some((code, _)) = account.code.scilla_code_and_init_data() else {
-        return Err(anyhow!("Address not contract address"));
+    let code = match account.code {
+        Code::Evm(ref bytes) => {
+            "EVM".to_string()
+                + &String::from_utf16(&bytes.into_iter().map(|v| *v as u16).collect::<Vec<_>>())
```

Comment:
Why utf16? Surely this will fail most of the time. `bytes` is usually not going to be a valid UTF-16 string.

Should we instead use `hex::encode`?

Reply:
ZQ1 returns this directly as a string, the only way I could get a result that is similar is by encoding each byte as a utf16 code point which allows all values of one byte whereas utf8 does not allow all one byte values

Reply:
Oh yeah
```
curl -X POST https://api.zilliqa.com/ -H 'content-type: application/json' -d '{"jsonrpc":"2.0","id":"1","method":"GetSmartContractCode","params":["e30161F32A019d876F082d9FF13ed451a03A2086"]}'
```
The current response is super cursed :)

I think we should make a breaking change here and return EVM code as a hex string. I don't think there's any need to prefix them with `"EVM"` either.

We should document this in `docs/api/zilliqa/getsmartcontractcode.doc.md`

Reply:
Surely there still needs to be a way to differentiate scilla code from evm here? Is it worth adding an extra field to the output such as `language` with `scilla` or `evm`?

Reply:
To be pedantic, EVM isn't a language per se, so I personally would call the field `runtime` rather than `language`

Reply:
`runtime`, `engine`, `type` all seem decent to me. I agree that `language` is misleading.